File: 2025-1/Clase25/STL_LIST/Persona.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   Persona.cpp
 * Author: hecto
 * 
 * Created on June 11, 2025, 4:51 PM
 */

#include <valarray>

#include "Persona.hpp"

Persona::Persona() {
}

Persona::Persona(const Persona& orig) {
    *this = orig;
}

Persona::~Persona() {
}

void Persona::SetSueldo(double sueldo) {
    this->sueldo = sueldo;
}

double Persona::GetSueldo() const {
    return sueldo;
}

void Persona::SetDni(int dni) {
    this->dni = dni;
}

int Persona::GetDni() const {
    return dni;
}

void Persona::SetNombre(string nombre) {
    this->nombre = nombre;
}

string Persona::GetNombre() const {
    return nombre;
}

void Persona::leer(){
    char buffer[50];
    cin>>dni;
    if(cin.eof())return;
    cin.get();
    cin.getline(buffer, 60, ',');
    nombre = buffer; // string -> char* ya tiene la sobrecarga del operador = dentro d sus especificacion
    cin>>sueldo;
    cin.get();
}

void Persona::imprimir() const{
    cout.precision(2);
    cout<<fixed;
    cout<<right<<setw(10)<<dni<<" "<<left<<setw(30)<<nombre;
    cout<<right<<setw(10)<<sueldo<<endl;
}

void Persona::operator =(const class Persona&p){
    nombre = p.nombre;
    dni = p.dni;
    sueldo = p.sueldo;
}

void operator>>(istream& input, class Persona& p){
    p.leer();
}

ostream& operator<<(ostream& output, const class Persona & p){
    p.imprimir();
    return output;
}

File: 2025-1/Clase29/SegundoProyecto/src/segundoproyecto/Persona.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package segundoproyecto;

/**
 *
 * @author hecto
 */
public class Persona {
    private int dni;
    private String nombre;
    private double sueldo;
    private boolean estado;
    private char sexo;

    //Constructores
    public Persona(){
        sueldo = 0.0;
        estado = false;
    }
    
    public Persona(int dni, String nombre, double sueldo, boolean estado, char sexo){
        this.dni = dni;
        this.nombre = nombre;
        this.sueldo = sueldo;
        this.estado = estado;
        this.sexo = sexo;
    }
    /**
     * @return the dni
     */
    public int getDni() {
        return dni;
    }

    /**
     * @param dni the dni to set
     */
    public void setDni(int dni) {
        this.dni = dni;
    }

    /**
     * @return the nombre
     */
    public String getNombre() {
        return nombre;
    }

    /**
     * @param nombre the nombre to set
     */
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    /**
     * @return the sueldo
     */
    public double getSueldo() {
        return sueldo;
    }

    /**
     * @param sueldo the sueldo to set
     */
    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }

    /**
     * @return the estado
     */
    public boolean isEstado() {
        return estado;
    }

    /**
     * @param estado the estado to set
     */
    public void setEstado(boolean estado) {
        this.estado = estado;
    }

    /**
     * @return the sexo
     */
    public char getSexo() {
        return sexo;
    }

    /**
     * @param sexo the sexo to set
     */
    public void setSexo(char sexo) {
        this.sexo = sexo;
    }
    
    public void imprimir(){
        System.out.println(dni + " - " + nombre + " - " + sueldo + " - " + sexo + " - " + estado);
        System.out.println();
    }
    
}
